src/crypto/PublicKey.ts

```diff
@@ -211,6 +217,118 @@ export class SignedPublicKey
   }
 }
 
+export class AccountLinkedPublicKeyV1
+  extends UnsignedPublicKey
+  implements publicKey.AccountLinkedPublicKey_V1
+{
+  keyBytes: Uint8Array
+  staticSignature: AccountLinkedStaticSignature | undefined
+  siweSignature: signature.AccountLinkedSIWESignature | undefined
+
+  constructor(obj: publicKey.AccountLinkedPublicKey_V1) {
+    if (!obj.keyBytes) {
+      throw new Error('Invalid AccountLinkedPublicKeyV1 has no keyBytes')
+    }
+    super(publicKey.UnsignedPublicKey.decode(obj.keyBytes))
+    this.keyBytes = obj.keyBytes
+    if (obj.staticSignature) {
+      this.staticSignature = new AccountLinkedStaticSignature(
+        obj.staticSignature
+      )
+    } else {
+      throw new Error(
+        'Unimplemented signature type for AccountLinkedPublicKeyV1'
+      )
+    }
+  }
+
+  // Return bytes of the encoded unsigned key.
+  private bytesToSign(): Uint8Array {
+    return this.keyBytes
+  }
+
+  public getLinkedAddress(role: AccountLinkedRole): string {
+    if (this.staticSignature) {
+      const expectedTextBytes = toUtf8Bytes(
+        WalletSigner.accountLinkRequestText(this.bytesToSign(), role)
+      )
+      if (!equalBytes(this.staticSignature.text, expectedTextBytes)) {
+        throw new Error('Signature text does not match expected text')
+      }
+      const digest = hexToBytes(utils.hashMessage(this.staticSignature.text))
+      const signature = this.staticSignature.walletEcdsaCompact
+      const publicKey = ecdsaSignerKey(digest, signature)
+      if (!publicKey) {
+        throw new Error('Unable to derive address from signature')
```

Comment:
Nice, I do like this error message a lot better. Still wondering if we should just keep an expected peerAddress with the AccountLinkedKey so we can actually validate.